Friendly-plans/app/src/androidTest/java/pg/autyzm/friendly_plans/matchers/ErrorTextMatcher.java

```diff
@@ -0,0 +1,37 @@
+package pg.autyzm.friendly_plans.matchers;
+
+import android.view.View;
+import android.widget.EditText;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+/**
+ * Use this custom matcher when you need match/assert <b>error text</b> on mandatory(*) EditText
+ * fields in xml layout.
+ */
+
+public class ErrorTextMatcher {
+
+    public static Matcher<View> hasErrorText(final String expectedErrorText) {
+        return new TypeSafeMatcher<View>() {
+
+            @Override
+            public boolean matchesSafely(View view) {
+                if (!(view instanceof EditText)) {
+                    return false;
+                }
+
+                CharSequence error = ((EditText) view).getError();
+                if (error == null) {
+                    return false;
+                }
+
+                String actualError = error.toString();
+                return expectedErrorText.equals(actualError);
+            }
+            @Override
+            public void describeTo(Description description) { }
```

Comment:
Codacy require method no to be empty and in that case it is true. Please update this method with you custom message for matcher. Description is here: http://www.planetgeek.ch/2012/03/07/create-your-own-matcher/